src/PolyhedralGeometry/Polyhedron/standard_constructions.jl

```diff
@@ -1187,14 +1159,14 @@ function rand_spherical_polytope(d::Int, n::Int; distribution::Symbol=:uniform,
   else
     throw(ArgumentError("rand_spherical_polytope: invalid distribution specified"))
   end
-  opts = Dict{Symbol,Any}( :template_parameters => [type] )
-  if seed != nothing
+  opts = Dict{Symbol,Any}( :template_parameters => [type] ) # creating the Optionset, the :template_parameters are for templating functions in C++
+  if !isnothing(seed)
     opts[:seed] = convert(Int64, seed)
   end
-  if precision != nothing
+  if !isnothing(precision)
     opts[:precision] = convert(Int64, precision)
   end
-  pm_obj = Polymake.call_function(:polytope, :rand_sphere, d, n; opts...)::Polymake.BigObject
+  pm_obj = Polymake.call_function(:polytope, :rand_sphere, d, n; opts...)::Polymake.BigObject # specifying the Type so it will throw an error if its not this type
```

Comment:
```suggestion
  pm_obj = Polymake.call_function(:polytope, :rand_sphere, d, n; opts...)::Polymake.BigObject
```
This is to tell the compiler the return type since `call_function` is not type-stable.